File: Signawel.Mobile/Signawel.Mobile/Constants/ApiConstants.cs
﻿using Signawel.Domain.Enums;

namespace Signawel.Mobile.Constants
{
    public static class ApiConstants
    {
        public static string BaseApiAddress => "https://signawel-api.azurewebsites.net/api/";

        #region Authentication

        public static string LoginEndpoint => BaseApiAddress + "authentication/login";

        public static string RegisterEndpoint => BaseApiAddress + "authentication/register";

        public static string RefreshEndpoint => BaseApiAddress + "authentication/refresh";

        #endregion

        #region DeterminationGraph

        public static string GetDeterminationGraph => BaseApiAddress + "determination-graph";

        public static string PostDeterminationGraph => BaseApiAddress + "determination-graph";

        #endregion

        #region Images

        public static string GetImage(string imageId) => $"{BaseApiAddress}images/{imageId}";

        public static string PostImage => BaseApiAddress + "images";

        #endregion

        #region Categories

        public static string GetCategories => $"{BaseApiAddress}categories";

        #endregion

        #region RoadworkSchemas

        public static string GetRoadworkSchema(string roadworkSchemaId) =>
            $"{BaseApiAddress}roadwork-schemas/{roadworkSchemaId}";

        public static string GetAllRoadworkSchemas => BaseApiAddress + "roadwork-schemas";

        public static string GetAllRoadworkSchemasByCategory(RoadworkCategory category) => $"{ BaseApiAddress }roadwork-schemas?roadworkCategory={ category }";

        public static string PostRoadworkSchema => BaseApiAddress + "roadwork-schemas";

        public static string PutRoadworkSchema(string roadworkSchemaId) =>
            $"{BaseApiAddress}roadwork-schemas/{roadworkSchemaId}";

        public static string DeleteRoadworkSchema(string roadworkSchemaId) =>
            $"{BaseApiAddress}roadwork-schemas/{roadworkSchemaId}";

        #endregion

        #region Reporting

        public static string PostReport => BaseApiAddress + "reports";

        #endregion

        #region Issues

        public static string GetDefaultIssues => BaseApiAddress + "issues/default";

        #endregion
    }
}

File: Signawel.Web/ClientApp/src/app/constants/api.constants.ts
export const BASE_URL = "https://localhost:5001/api";
export const AUTHENTICATE_URL = BASE_URL + "/authentication";
export const AUTHENTICATE_LOGIN_URL = AUTHENTICATE_URL + "/login";
export const AUTHENTICATE_REFRESH_URL = AUTHENTICATE_URL + "/refresh";
export const    AUTHENTICATE_CONFIRMEMAIL_URL = AUTHENTICATE_URL + "/confirmemail";
export const IMAGES = BASE_URL + "/images";
export const DETERMINATION_GRAPH = BASE_URL + "/determination-graph";
export const ROADWORK_SCHEMAS = BASE_URL + "/roadwork-schemas";
export const REPORT_GROUPS = BASE_URL + "/reportgroups";
export const CITIES = REPORT_GROUPS + "/cities";
export const PRIORITY_EMAILS = BASE_URL + "/priority-emails";
export const REPORTS = BASE_URL + "/reports"

// GIPOD
export const GIPOD_BASE_URL = 'http://api.gipod.vlaanderen.be/ws/v1';
export const ROADWORK_INFO = GIPOD_BASE_URL + '/WorkAssignment?Id=';